src/clinical/service-worker-thread/tasks.ts

```diff
@@ -213,6 +216,35 @@ const mapEntityDocuments = (
   };
 };
 
+export function filterDonorIdDataFromSearch(donors: Donor[], query: ClinicalSearchQuery) {
+  const { donorIds, submitterDonorIds } = query;
+
+  const useFilteredDonors =
+    (donorIds && donorIds.length) || (submitterDonorIds && submitterDonorIds.length);
+
+  const filteredDonors = useFilteredDonors
+    ? donors.filter(donor => {
+        const { donorId, submitterId } = donor;
+        const stringId = `${donorId}`;
+        const donorMatch = donorIds?.filter(id => stringId.includes(id));
+        const submitterMatch = submitterId
+          ? submitterDonorIds?.filter(id => submitterId.includes(id))
+          : [];
+        return donorMatch.length > 0 || submitterMatch.length > 0;
+      })
+    : donors;
```

Comment:
based on the typing of `ClinicalSearchQuery` both `donorIds` and `submitterDonorIds` will always at least be an empty array, so once this is an accurate reflection (eg. default value is accounted for) , there's no need for the optional chaining as it's already an `[]` and not potentially `undefined`

Reply:
looks like they may be undefined coming from frontend though https://github.com/icgc-argo/platform-ui/blob/develop/generated/gql_types.tsx#L163